### DISCUSSION TITLE
[solved] Transaction on TestNet-GoChain in python (web3.py) doesn't work

### DISCUSSION THREAD

**Discussion by er1202** [gochain-io/docs#103] [Q&A]
Hello,
i try to make a GO transaction on the gochain-testnet with web3.py
Here is my code:

`rpcURL = 'https://testnet-rpc.gochain.io'
web3 = Web3(Web3.HTTPProvider(rpcURL))
print(web3.isConnected())

transaction = {'to': MetaMaskAddress,
               'value': web3.toWei(5, 'ether'), 'nonce': 22,
               'from': testWalletAddress,
               'gas': 21000, 'gasPrice': web3.eth.gas_price}

signed = web3.eth.account.sign_transaction(transaction, testPrivatKey)

print(web3.toHex(web3.eth.send_raw_transaction(signed.rawTransaction)))
`
This gives me no error and a Transaction hash: 
(for example: _0xf8c6cf696549a2411fcabfe6f349e04d67337bc5ab547b6066654267242cb16f_) but it doesn't send the (test-)GO and i can't see the transaction on test-gochain explorer as well.

Has anyone an idea what i'm doing wrong.... (Maybe an issue with nonce or gastric?)

if i try the same with the gochain/web3 client it works without problems

`export WEB3_NETWORK=testnet
export WEB3_PRIVATE_KEY=0x88a741c44900bxxxx
web3 transfer 3 to 0x4Ca8606A73xxxxx
`

**Comment by er1202** [comment]
Solved the problem, simple changed the nonce definition to:
nonce = web3.eth.getTransactionCount(testWalletAddress)

**Reply by treeder** [reply]
You can usually leave the nonce out and it will do that automatically in most web3 libraries.
